flare/framework/computation_task.py

```diff
@@ -26,39 +29,40 @@ class ComputationTask(object):
     c. define a ComputationTask with the algorithm
     """
 
-    def __init__(self, algorithm):
+    def __init__(self, algorithm, hyperparas=dict(lr=1e-4)):
         assert isinstance(algorithm, Algorithm)
+        self.hp = hyperparas
         self.alg = algorithm
-        if torch.cuda.is_available() and self.alg.gpu_id >= 0:
-            self.device = torch.device("cuda:" + str(self.alg.gpu_id))
-        else:
-            self.device = torch.device("cpu")
-        ## put the model on the device
-        self.alg.model.to(self.device)
+        self.optim = optim.RMSprop(
+            self.alg.model.parameters(), lr=hyperparas["lr"])
 
-    def _create_tensors(self, np_arrays_dict, specs):
-        ## We want to convert numpy arrays to torch tensors,
+    def _create_tensors(self, arrays_dict, specs):
+        ## We want to convert python arrays to a hierarchy of torch tensors,
         ## and put them on the device
         tensors = {}
         for name, props in specs:
-            assert name in np_arrays_dict, "keyword %s does not exist in np arrays!" % name
-
-            np_array = np_arrays_dict[name]
-            assert isinstance(np_array, np.ndarray)
+            assert name in arrays_dict, "keyword %s does not exist in python arrays!" % name
+            array = arrays_dict[name]
             dtype = ("float32" if "dtype" not in props else props["dtype"])
-            assert np_array.dtype == dtype, "%s %s" % (np_array.dtype, dtype)
             assert "shape" in props, "You must specify the tensor shape in the specs!"
```

Comment:
might be better to also add documents/checks on this type of requirements (e.g. assert "shape" in props) to a place closer to the place where specs is created, .e.g get_input_specs() in model or algo.